gapipy/models/base.py

```diff
@@ -69,6 +72,17 @@ def _fill_fields(self, data):
             elif field in first(self._resource_collection_fields):
                 self._set_resource_collection_field(field, value)
 
+        # overide uri for unconventional resources (ie oauth/token)
+        self._set_uri_override_field()
+
+    def _override_id_field(self):
+        if hasattr(self, '_id_lookup_field'):
```

Comment:
Another option for this is to always have an `_id_lookup_field`, but the default defined is `id`

Thus, no surprises. `self.id` is always sourced from `getattr(self, self._id_lookup_field)`, and the Token class is safe to modify that field.

Basically, very close to what you're doing but just assumes that class attribute is always there -- We can set it on the base model.